### ISSUE TITLE
[BUG] _embedding_bag_dense_backward: mode=2 (max) silently receives the mean gradient — maximum_indices is accepted and never used

### ISSUE THREAD

**Issue by truong-v** [FlagOpen/FlagGems#5220]
## Summary

`_embedding_bag_dense_backward_kernel` has two arms: `mode == 0` applies the bag gradient as is (correct for sum), everything else divides it by the bag size (correct for mean) — and `mode == 2` (max) falls into that else. Max pooling's backward is different in kind: for each feature the whole gradient of a bag belongs to the single row that won the max, and every other row gets nothing. The wrapper even accepts `maximum_indices` — the only argument that says which row won — and never forwards it to any launch.

Under `flag_gems.enable()`, a plain `nn.EmbeddingBag(..., mode="max")` backward on CUDA trains on wrong gradients with no error or warning. `tests/test_embedding_bag_dense_backward.py` drives modes 0 and 1 only.

The kernel was added in #3900.

## Reproduction

```python
import torch

import flag_gems

torch.manual_seed(0)

weight = torch.randn(5, 3, device="cuda")
indices = torch.tensor([0, 1, 2, 3, 4, 0], device="cuda")
offsets = torch.tensor([0, 3], device="cuda")

# Real forward (mode 2 = max) supplies offset2bag / bag_size / maximum_indices.
out, offset2bag, bag_size, max_indices = torch.ops.aten._embedding_bag(
    weight, indices, offsets, False, 2, False, None, False, -1
)
go = torch.randn_like(out)

ref = torch.ops.aten._embedding_bag_dense_backward(
    go, indices, offset2bag, bag_size, max_indices, 5, False, 2, None, -1
)
with flag_gems.use_gems():
    res = torch.ops.aten._embedding_bag_dense_backward(
        go, indices, offset2bag, bag_size, max_indices, 5, False, 2, None, -1
    )

print("mode=2 (max):  max|gems - aten| =", (res - ref).abs().max().item())
print("aten grad_weight:")
print(ref.cpu().numpy().round(4))
print("flag_gems grad_weight:")
print(res.cpu().numpy().round(4))
```

```
mode=2 (max):  max|gems - aten| = 0.9199963808059692
aten grad_weight:
[[ 0.      0.      0.    ]
 [ 0.1808 -0.5523  0.9238]
 [ 0.      0.      0.    ]
 [ 0.      0.      0.8676]
 [-0.735   1.38    0.    ]]
flag_gems grad_weight:
[[-0.1847  0.2759  0.5971]
 [ 0.0603 -0.1841  0.3079]
 [ 0.0603 -0.1841  0.3079]
 [-0.245   0.46    0.2892]
 [-0.245   0.46    0.2892]]
```

Every cell differs; three rows that must be exactly zero receive gradient. Through the module path, `nn.EmbeddingBag(5, 3, mode="max")` backward under `use_gems()` differs from eager by 1.78 max abs on `weight.grad`. Modes 0 and 1 match eager exactly (verified as controls).

## Environment

- FlagGems master (75cf3ea)
- NVIDIA L40S
- torch 2.6.0+cu124, triton 3.2.0
